### PR TITLE
Fix leaderboard bug and added --extra option.

### PR DESCRIPTION
**Pull request by NicEastvillage**
- Fixed leaderboard bug involving new bots being placed above moving bots
- ~~Added the `--extra` option to show five more divisions~~
- Added colors for the remaining divisions (just grey)
Edit:
- Now automatically adds another column when bot count is greater than 40
- The `--top-only` option will force it to only add two columns, top 40, even though there might be more bots

### CODE REVIEW

**Comment by skyborgff** [comment]
cant -- extra be auto activated if there are more bots than the one sthat fit normally, and use --standard for showing only the first colored divisions?

**Comment by NicEastvillage** [comment]
You are right, that is probably better. I was tired when I made this ;)

**Comment by NicEastvillage** [comment]
It now behaves as @skyborgff described, except I called the option for `--top-only`.

File: autoleagueplay/leaderboard/leaderboard.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ def generate_leaderboard(working_dir: WorkingDir, run_strategy: RunStrategy, ext
     # Divisions. We only have color palettes configured for a certain number of them, so enforce a limit.
     divisions = Ladder.DIVISION_NAMES[:len(Symbols.palette)]
 
+    extra = (allow_extra and len(new_ladder.bots) > 40)
```
Comment (Sciguymjm) [inline autoleagueplay/leaderboard/leaderboard.py]: What happens when there are 61 bots?

Comment (NicEastvillage) [reply autoleagueplay/leaderboard/leaderboard.py]: Those will be drawn off screen.

When/if that time comes we should make it automatically expand with more columns, or decide to only draw the top bots.

**Pull request merged**